src/microbithal_js.c

```diff
@@ -113,37 +113,11 @@ int microbit_hal_pin_set_analog_period_us(int pin, int period) {
         mp_js_hal_audio_period_us(period);
         return 0;
     }
-
-    /*
-    // Calling setAnalogPeriodUs requires the pin to be in analog-out mode.  So
-    // test for this mode by first calling getAnalogPeriodUs, and if it fails then
-    // attempt to configure the pin in analog-out mode by calling setAnalogValue.
-    if ((ErrorCode)pin_obj[pin]->getAnalogPeriodUs() == DEVICE_NOT_SUPPORTED) {
-        if (pin_obj[pin]->setAnalogValue(0) != DEVICE_OK) {
-            return -1;
```

Comment:
No matter what I do I can't get this or the error branch below to trigger on a V2 device. So for now, the simulator calls always succeed. 

@dpgeorge do you know how I can trigger these error branches?

Link to easier to read version: https://github.com/microbit-foundation/micropython-microbit-v2/blob/e4321a866e3f8ef48a7f5b5e96db99cc39b09757/src/codal_app/microbithal.cpp#L155